lxd/db/cluster/secrets.go

```
@@ -0,0 +1,84 @@
+package cluster
+
+import (
+	"crypto/rand"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+)
+
+// Secret contains a 512 byte slice and a creation time (in unix milliseconds).
+type Secret struct {
+	Value     []byte `json:"value"`
+	CreatedAt int64  `json:"created_at"`
+}
+
+// IsValid returns true if the Secret is younger than the given lifetime.
+func (s Secret) IsValid(lifetime int64) bool {
+	return s.CreatedAt+lifetime > time.Now().UTC().UnixMilli()
+}
+
+// Secrets is a concrete type for a slice of Secret. This allows defining convenient methods.
+type Secrets []Secret
+
+// IsValid returns true if the first Secret in Secrets is younger than the given lifetime.
+func (s Secrets) IsValid(lifetime int64) bool {
+	return len(s) > 0 && s[0].IsValid(lifetime)
+}
+
+// Rotate prepends a new secret. Currently, the slice length is limited to 2.
+func (s *Secrets) Rotate() {
+	secretValue := make([]byte, 512)
+	_, _ = rand.Read(secretValue)
```

I saw the `DeviceNextInterfaceHWAddr` func in lxd/network/network_utils.go return and handle the error from crypto/rand, so would be better to handle err here too?

---

I ignored the error here per the golang docs https://pkg.go.dev/crypto/rand#Read

> Read fills b with cryptographically secure random bytes. It never returns an error, and always fills b entirely.